### ISSUE TITLE
feature: dedicated Error type for ClientResult

### ISSUE THREAD

**Issue by fussybeaver** [abdolence/slack-morphism-rust#59]
Error handling is much nicer now with callbacks falling back to the error_handler, thanks for that. 

It'd also be great ergonomics if the errors from the slack client have a dedicated type, because it allows consumers to write error handling enums that handle those errors. At the moment the slack client will always return a ClientResult, which just has a boxed trait object std error: https://docs.rs/slack-morphism/0.14.0/slack_morphism/type.ClientResult.html

For example:
```
client.open_session(...).conversations_create(...).await? // <- Result<SlackApiConversationsCreateResponse, Box<dyn std::error::Error>> instead of, say Result<SlackApiConversationsCreateResponse, SlackClientError>
```

**Comment by abdolence** [comment]
I agree, explicit types for the win when possible. Let me check if I can improve this easily.

**Comment by abdolence** [comment]
Fixed with https://github.com/abdolence/slack-morphism-rust/pull/60

**Comment by abdolence** [comment]
Released in v0.15.0.
